### PR TITLE
get_mol_kegg() to replace save_compound_mol()

### PR DESCRIPTION
**Pull request by Aariq**
A new, vectorized function for downloading mol files from KEGG given vectors of compound IDs or pathway IDs.  Returns a tibble including the IDs and the locations of the saved .mol files for use as input (eventually) to `calc_vol()`

One additional feature this has is that it queries the compound name and adds that to the title line of the .mol file.  This could be used later by in `calc_vol()` to add compound name to the output rather than querying KEGG.

### CODE REVIEW

**Comment by Aariq** [comment]
Ok, this is _actually_ ready for review now.

File: R/get_mol_kegg.R [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#' Download compound .mol files from KEGG
+#'
+#' Downloads mol files corresponding to individual compounds or compounds in a
+#' pathway from KEGG.
+#'
+#' @param compound_ids character vector of KEGG compound IDs---5 digits
+#'   prepended with a "C".
+#' @param pathway_ids character vector of KEGG pathway IDs---5 digits prepended
+#'   with "map".
+#' @param dir path to a folder to save mol files in.
```
Comment (KristinaRiemer) [inline R/get_mol_kegg.R:11]:
```suggestion
#' @param dir path to a folder to save mol files in. Folder will be created if it does not already exist
```

File: R/get_mol_kegg.R [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    names(compound_ids_list) <- pathway_ids
+    out_tbl <- 
+      tibble::enframe(compound_ids_list, name = "pathway_id", value = "compound_id") %>% 
+      tidyr::unnest(tidyselect::everything()) %>% 
+      dplyr::mutate(mol_path = fs::path(dir, .data$pathway_id, .data$compound_id, ext = "mol"))
+    
+  }
+  
+  # Download mols
+  .get_mol_kegg <- function(compound_id) {
```
Comment (KristinaRiemer) [inline R/get_mol_kegg.R:58]: @Aariq could you explain to me what the function name with the period before it is doing?

File: R/get_mol_kegg.R [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    #TODO I think the KEGG API can handle up to 10 requests at once separated by
```
Comment (KristinaRiemer) [inline R/get_mol_kegg.R:59]: I got a timeout, is that related to this API limit? 
```
> ex_pathway <- get_mol_kegg(pathway_ids = "map00253", dir = tempdir())
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: [rest.kegg.jp] Operation timed out after 10002 milliseconds with 0 out of 0 bytes received
```

**Review by KristinaRiemer — COMMENTED** [review]
First part of my review, might be more to come!

File: R/get_mol_kegg.R (hunk 2 above)
Comment (Aariq) [reply R/get_mol_kegg.R:58]: The function `.get_mol_kegg()` gets a .mol file for a single `compound_id`.  The enclosing `get_mol_kegg()` function vectorizes this (line 73).  Can't remember where, but I've seen this pattern before in R packages that are like:

``` r
foo <- function(vec) {
  .foo <- function(x){
    ...
  }
  lapply(vec, .foo)
}
```

But `.get_mol_kegg` could be called anything.

File: R/get_mol_kegg.R (hunk 3 above)
Comment (Aariq) [reply R/get_mol_kegg.R:59]: maybe?  I've gotten occasional timeouts as well, but usually just running the function a second time works.  It would be good to handle this TODO eventually (like, before CRAN submission maybe) but I think it works as-is for now.  If you get time-outs consistently I'll take a closer look.

Comment (Aariq) [reply R/get_mol_kegg.R:60]: I added a `httr2::req_retry(max_tries = 3)` which should help with this issue
